lib/spack/spack/test/concretize.py

```diff
@@ -1953,3 +1956,62 @@ def test_external_python_extensions_have_dependency(self):
 
         assert "python" in spec["py-extension1"]
         assert spec["python"] == spec["py-extension1"]["python"]
+
+    def test_external_python_extension_find_dependency_from_config(self):
+        external_conf = {
+            "py-extension1": {
+                "buildable": False,
+                "externals": [{"spec": "py-extension1@2.0", "prefix": "/fake"}],
+            },
+            "python": {
+                "externals": [{"spec": "python@configured", "prefix": "/fake"}],
+            },
+        }
+        spack.config.set("packages", external_conf)
+
+        spec = Spec("py-extension1").concretized()
+
+        assert "python" in spec["py-extension1"]
+        assert spec["python"].prefix == "/fake"
+        # The spec is not equal to spack.spec.Spec("python@configured") because it gets a
+        # namespace and an external prefix before marking concrete
+        assert spec["python"].satisfies("python@configured")
```

Comment:
At least one other scenario that might be worth testing:

* A->Bextension->python, A->python, there is an external python but A doesn't use it (because it demands a different version). Make sure Bextension uses the same Python as A.

I need to think more about whether more use cases would be helpful for ensuring coverage of this

Reply:
If an extension E is in the same solve as python P, E will use P. Regardless of what else we can say about the structure.

My inclination is that we don't need to create every structure we can think of, because understanding the code makes clear that all these structures will be handled identically (which is definitely a change from earlier iterations of this PR). If you feel strongly that we need to add more tests or that I'm wrong about this, let me know.